instrumentation/jetty/jetty-11.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/v11_0/Jetty11HandlerAdvice.java

```diff
@@ -31,6 +32,8 @@ public static void onEnter(
 
     context = tracer().startServerSpan(request);
     scope = context.makeCurrent();
+
+    tracer().setAsyncListenerResponse(request, response);
```

Comment:
do we need to add calls to `setAsyncListenerResponse` in each app server instrumentation, or can we do it once in servlet instrumentation?

Reply:
Since it is possible for requests to be handled entirely within app server handlers without servlet advices ever triggering - yes.

Reply:
can those requests have servlet async listeners?

Reply:
Jetty handlers and Tomcat valves both expose the `ServletRequest` and are allowed to generate the response without a servlet ever being invoked. I have not tested using `startAsync` in them, but the documentation does not say that either of them would only support synchronous responses.

Reply:
can you add comments to the code explaining this? otherwise worried these lines could be removed thinking they aren't really needed, and I don't think we have any tests that it would break

Reply:
Added those comments for all the appservers.